src/sentry/api/serializers/rest_framework/dashboard.py

```diff
@@ -144,6 +167,22 @@ def validate_interval(self, interval):
         return interval
 
     def validate(self, data):
+        is_issue_widget = data.get("widget_type") == DashboardWidgetTypes.ISSUE
+        query_errors = []
+        has_query_error = False
+        if data.get("queries"):
+            # Check each query to see if they have an issue or discover error depending on the type of the widget
+            for query in data.get("queries"):
+                if is_issue_widget and "issue_query_error" in query:
+                    query_errors.append(query["issue_query_error"])
+                    has_query_error = True
+                elif not is_issue_widget and "discover_query_error" in query:
+                    query_errors.append(query["discover_query_error"])
+                    has_query_error = True
+                else:
+                    query_errors.append({})
```

Comment:
What does the empty dict do?

Reply:
Hey Mark, some widgets can have multiple queries, each can independently be valid or invalid. For valid widgets, I append `{}` to indicate no error to the `query_errors` list since the order of this list matters in the frontend when mapping a query error to the proper query field ui.